reporting/reporter.go

```diff
@@ -0,0 +1,274 @@
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".max"), float64(h.Max()), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".mean"), h.Mean(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".std-dev"), h.StdDev(), 0, r.source, tags)
+	for psIdx, psKey := range r.percentiles {
+		key := strings.Replace(strconv.FormatFloat(psKey*100.0, 'f', -1, 64), ".", "", 1)
+		r.errors <- r.sender.SendMetric(r.prepareName(name+"."+key+"-percentile"), ps[psIdx], 0, r.source, tags)
+	}
+}
+
+func (r *reporter) reportMeter(name string, metric metrics.Meter, tags map[string]string) {
+	m := metric.Snapshot()
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".count"), float64(m.Count()), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".one-minute"), m.Rate1(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".five-minute"), m.Rate5(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".fifteen-minute"), m.Rate15(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".mean"), m.RateMean(), 0, r.source, tags)
+}
+
+func (r *reporter) reportTimer(name string, metric metrics.Timer, tags map[string]string) {
+	t := metric.Snapshot()
+	du := float64(r.durationUnit)
+	ps := t.Percentiles(r.percentiles)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".count"), float64(t.Count()), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".min"), float64(t.Min()/int64(du)), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".max"), float64(t.Max()/int64(du)), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".mean"), t.Mean()/du, 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".std-dev"), t.StdDev()/du, 0, r.source, tags)
+	for psIdx, psKey := range r.percentiles {
+		key := strings.Replace(strconv.FormatFloat(psKey*100.0, 'f', -1, 64), ".", "", 1)
+		r.errors <- r.sender.SendMetric(r.prepareName(name+"."+key+"-percentile"), ps[psIdx]/du, 0, r.source, tags)
+	}
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".one-minute"), t.Rate1(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".five-minute"), t.Rate5(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".fifteen-minute"), t.Rate15(), 0, r.source, tags)
+	r.errors <- r.sender.SendMetric(r.prepareName(name+".mean-rate"), t.RateMean(), 0, r.source, tags)
+}
+
+func (r *reporter) prepareName(name string, suffix ...string) string {
+	if len(r.prefix) > 0 {
+		name = r.prefix + "." + name
+	}
+
+	if r.addSuffix {
+		for _, s := range suffix {
+			name += "." + s
+		}
+	}
+
+	return name
+}
+
+func (r *reporter) Start() {
+	r.running = true
+}
+
+func (r *reporter) Stop() {
+	r.running = false
+	r.Report()
+}
+
+func (r *reporter) Close() {
+	r.Report()
```

Comment:
Should also close the sender.

Reply:
the sender is created outside... so it should be close outside, just in case is use to other thing.